source/io/module_httpClient.js

```diff
@@ -760,24 +870,35 @@
             };
 
             httpClient.createRequest(requestData, function (responseData) {
-                Module.eggShell.dataWriteString(headersPointer, responseData.header);
-                Module.eggShell.dataWriteUInt32(statusCodePointer, responseData.status);
+                Module.eggShell.writeString(headersValueRef, responseData.header);
+                Module.eggShell.writeDouble(statusCodeValueRef, responseData.status);
 
-                if (bodyPointer !== NULL) {
-                    Module.eggShell.dataWriteStringFromArray(bodyPointer, responseData.body);
+                if (bodyValueRef !== undefined) {
+                    Module.eggShell.writeStringFromArray(bodyValueRef, responseData.body);
                 }
 
-                var newErrorStatus = responseData.labviewCode !== ERRORS.NO_ERROR.CODE;
-                var newErrorCode = responseData.labviewCode;
-                errorMessage = Module.coreHelpers.formatMessageWithException(responseData.errorMessage, responseData.requestException);
-                var newErrorSource = Module.coreHelpers.createSourceFromMessage(errorMessage);
-                Module.coreHelpers.mergeErrors(newErrorStatus, newErrorCode, newErrorSource, errorStatusPointer, errorCodePointer, errorSourcePointer);
+                var errorMessage = Module.coreHelpers.formatMessageWithException(responseData.errorMessage, responseData.requestException);
+                var newError = {
+                    status: responseData.labviewCode !== ERRORS.NO_ERROR.CODE,
+                    code: responseData.labviewCode,
+                    source: Module.coreHelpers.createSourceFromMessage(errorMessage)
+                };
+
+                Module.coreHelpers.mergeErrors(errorValueRef, newError);
                 Module.eggShell.setOccurrenceAsync(occurrencePointer);
             });
         };
 
-        Module.httpClient.jsHttpClientConfigCORS = function (handle, includeCredentialsDuringCORS, errorStatusPointer, errorCodePointer, errorSourcePointer) {
-            var httpClient = findhttpClientOrWriteError(handle, errorStatusPointer, errorCodePointer, errorSourcePointer);
+        Module.httpClient.jsHttpClientConfigCORS = function (
+            handleTypeRef, handleDataRef,
+            includeCredentialsDuringCORSTypeRef, includeCredentialsDuringCORSDataRef,
+            errorTypeRef, errorDataRef) {
+            var handleValueRef = Module.eggShell.createValueRef(handleTypeRef, handleDataRef);
+            var includeCredentialsDuringCORSValueRef = Module.eggShell.createValueRef(includeCredentialsDuringCORSTypeRef, includeCredentialsDuringCORSDataRef);
+            var errorValueRef = Module.eggShell.createValueRef(errorTypeRef, errorDataRef);
+            var handle = Module.eggShell.readDouble(handleValueRef);
+            var includeCredentialsDuringCORS = Module.eggShell.readDouble(includeCredentialsDuringCORSValueRef);
```

Comment:
Can we move the !== FALSE check up here? That way includeCredentialsDuringCORS can just be a boolean

Reply:
ah yeah, I'll change it.